diff --git a/dist/cjs/index.cjs b/dist/cjs/index.cjs
index e470032a9572b3ced764ca02238a8c6be435a9d4..47370bb39e710198ed36af156150d290bdd09a41 100644
--- a/dist/cjs/index.cjs
+++ b/dist/cjs/index.cjs
@@ -470,9 +470,15 @@ class Virtualizer {
             onTouchEnd,
             addEventListenerOptions
           );
+          scrollEl.addEventListener(
+            "touchcancel",
+            onTouchEnd,
+            addEventListenerOptions
+          );
           this.unsubs.push(() => {
             scrollEl.removeEventListener("touchstart", onTouchStart);
             scrollEl.removeEventListener("touchend", onTouchEnd);
+            scrollEl.removeEventListener("touchcancel", onTouchEnd);
             if (this._iosTouchEndTimerId !== null && this.targetWindow != null) {
               this.targetWindow.clearTimeout(this._iosTouchEndTimerId);
               this._iosTouchEndTimerId = null;
diff --git a/dist/esm/index.js b/dist/esm/index.js
index 2495b26cf2c3589213546b3958eaadf2eb6b751d..32c1bc32b9a1b68bef77f8aeca8bb8f5ab3134bc 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -468,9 +468,15 @@ class Virtualizer {
             onTouchEnd,
             addEventListenerOptions
           );
+          scrollEl.addEventListener(
+            "touchcancel",
+            onTouchEnd,
+            addEventListenerOptions
+          );
           this.unsubs.push(() => {
             scrollEl.removeEventListener("touchstart", onTouchStart);
             scrollEl.removeEventListener("touchend", onTouchEnd);
+            scrollEl.removeEventListener("touchcancel", onTouchEnd);
             if (this._iosTouchEndTimerId !== null && this.targetWindow != null) {
               this.targetWindow.clearTimeout(this._iosTouchEndTimerId);
               this._iosTouchEndTimerId = null;
